### ISSUE TITLE
[WP Migration Phase 7] Assets & Optimization - Performance & Compatibility

### ISSUE THREAD

**Issue by luciana-razorvision** [razorvision/safequote.io#21]
## Phase 7: Assets & Optimization

Optimize assets, improve performance, and ensure compatibility with WordPress ecosystem.

#### Asset Minification
- [ ] Set up CSS minification (PostCSS cssnano)
- [ ] Set up JavaScript minification (Terser)
- [ ] Create production build script
- [ ] Add source maps for debugging
- [ ] Implement cache busting (version numbers)

#### Image Optimization
- [ ] Convert all images to WebP (with fallbacks)
- [ ] Create responsive image sizes
- [ ] Implement lazy loading for images
- [ ] Add proper alt text to all images
- [ ] Optimize hero images (compress, resize)
- [ ] Create image sprite for icons (if applicable)

#### Performance Optimization
- [ ] Conditional script/style loading (only on needed pages)
- [ ] Defer non-critical JavaScript
- [ ] Inline critical CSS
- [ ] Minimize HTTP requests
- [ ] Enable browser caching (via headers)
- [ ] Implement WordPress transients for expensive operations

#### Icon System
- [ ] Convert Lucide React icons to SVG sprite or inline SVGs
- [ ] Create icon helper function
- [ ] Implement icon caching
- [ ] Ensure icons are accessible (aria-hidden, role, etc.)

#### Font Optimization
- [ ] Use system fonts or optimized Google Fonts
- [ ] Implement font-display: swap
- [ ] Preload critical fonts
- [ ] Subset fonts (only needed characters)

#### JavaScript Optimization
- [ ] Remove unused code (tree shaking)
- [ ] Code splitting (separate vendor from custom code)
- [ ] Async loading for non-critical scripts
- [ ] Use event delegation for better performance
- [ ] Debounce/throttle scroll and resize events

#### CSS Optimization
- [ ] Purge unused Tailwind CSS classes
- [ ] Combine and minify stylesheets
- [ ] Remove duplicate CSS rules
- [ ] Optimize selector specificity
- [ ] Use CSS containment where appropriate

#### WordPress Compatibility Testing
- [ ] Test with Yoast SEO plugin
- [ ] Test with Rank Math plugin
- [ ] Test with Contact Form 7
- [ ] Test with WPForms
- [ ] Test with W3 Total Cache
- [ ] Test with WP Super Cache
- [ ] Test with Autoptimize plugin
- [ ] Test with popular page builders (Elementor, Beaver Builder)

#### Browser Compatibility
- [ ] Test in Chrome (latest 2 versions)
- [ ] Test in Firefox (latest 2 versions)
- [ ] Test in Safari (latest 2 versions)
- [ ] Test in Edge (latest 2 versions)
- [ ] Test on mobile browsers (iOS Safari, Chrome Mobile)
- [ ] Add polyfills if needed (Intersection Observer, fetch)

#### Mobile Optimization
- [ ] Test responsive design on all breakpoints
- [ ] Optimize touch targets (minimum 44x44px)
- [ ] Test mobile menu functionality
- [ ] Optimize images for mobile (srcset)
- [ ] Test form inputs on mobile
- [ ] Ensure horizontal scrolling is prevented

#### Accessibility Testing
- [ ] Run WAVE accessibility checker
- [ ] Test with screen reader (NVDA, VoiceOver)
- [ ] Ensure keyboard navigation works
- [ ] Check color contrast (WCAG AA)
- [ ] Add skip links
- [ ] Test form labels and error messages

#### Security Hardening
- [ ] Sanitize all user inputs
- [ ] Escape all outputs
- [ ] Use nonces for all forms
- [ ] Prevent SQL injection (use wpdb prepared statements)
- [ ] Validate and sanitize AJAX requests
- [ ] Remove WordPress version from head
- [ ] Disable file editing in admin (if appropriate)

#### PostCSS Configuration (postcss.config.js)
```javascript
module.exports = {
  plugins: [
    require('tailwindcss'),
    require('autoprefixer'),
    process.env.NODE_ENV === 'production' && require('cssnano')({
      preset: ['default', {
        discardComments: { removeAll: true },
        normalizeWhitespace: true,
        colormin: true,
      }]
    }),
  ].filter(Boolean)
};
```

#### Tailwind CSS Purge Configuration
```javascript
// tailwind.config.js
module.exports = {
  content: [
    './wp-content/themes/safequote-traditional/**/*.php',
    './wp-content/themes/safequote-traditional/**/*.js',
  ],
  theme: {
    // ...
  },
  plugins: [],
};
```

#### Production Build Script
```json
// package.json
{
  "scripts": {
    "build:css": "NODE_ENV=production postcss src/css/main.css -o wp-content/themes/safequote-traditional/assets/css/main.css",
    "build:js": "terser assets/js/*.js --compress --mangle -o wp-content/themes/safequote-traditional/assets/js/main.min.js",
    "build": "npm run build:css && npm run build:js",
    "watch:css": "postcss src/css/main.css -o wp-content/themes/safequote-traditional/assets/css/main.css --watch",
    "watch": "npm run watch:css"
  }
}
```

#### Conditional Script Loading
```php
// functions.php
function safequote_enqueue_assets() {
    // Main CSS (always loaded)
    wp_enqueue_style(
        'safequote-main',
        get_template_directory_uri() . '/assets/css/main.css',
        [],
        filemtime(get_template_directory() . '/assets/css/main.css')
    );

    // Main JS (always loaded)
    wp_enqueue_script(
        'safequote-main',
        get_template_directory_uri() . '/assets/js/main.min.js',
        [],
        filemtime(get_template_directory() . '/assets/js/main.min.js'),
        true
    );

    // Filters JS (only on pages with vehicle filtering)
    if (is_front_page() || is_page_template('page-safety-ratings.php')) {
        wp_enqueue_script(
            'safequote-filters',
            get_template_directory_uri() . '/assets/js/filters.min.js',
            ['safequote-main'],
            filemtime(get_template_directory() . '/assets/js/filters.min.js'),
            true
        );
    }

    // NHTSA API (only on safety ratings page)
    if (is_page_template('page-safety-ratings.php')) {
        wp_enqueue_script(
            'safequote-nhtsa',
            get_template_directory_uri() . '/assets/js/nhtsa-api.min.js',
            ['safequote-main'],
            filemtime(get_template_directory() . '/assets/js/nhtsa-api.min.js'),
            true
        );
    }

    // Localize script
    wp_localize_script('safequote-main', 'safequote', [
        'ajax_url' => admin_url('admin-ajax.php'),
        'nonce' => wp_create_nonce('safequote_nonce'),
    ]);
}
add_action('wp_enqueue_scripts', 'safequote_enqueue_assets');
```

#### Lazy Loading Images
```php
// template-parts/vehicle-card.php
<img
    src="<?php echo esc_url($vehicle['image']); ?>"
    srcset="<?php echo esc_attr(safequote_get_srcset($vehicle['image'])); ?>"
    sizes="(max-width: 768px) 100vw, (max-width: 1024px) 50vw, 33vw"
    alt="<?php echo esc_attr($vehicle['make'] . ' ' . $vehicle['model']); ?>"
    loading="lazy"
    width="400"
    height="300"
>
```

#### WordPress Transient Caching
```php
// Cache NHTSA API results
function safequote_get_cached_safety_rating($year, $make, $model) {
    $cache_key = 'nhtsa_' . $year . '_' . $make . '_' . $model;
    $cached = get_transient($cache_key);

    if ($cached !== false) {
        return $cached;
    }

    // Fetch from API (would be done via AJAX in reality)
    $data = safequote_fetch_nhtsa_data($year, $make, $model);

    // Cache for 1 week
    set_transient($cache_key, $data, WEEK_IN_SECONDS);

    return $data;
}
```

### Performance Targets
- Lighthouse Performance Score: 90+
- Lighthouse Accessibility Score: 95+
- Lighthouse Best Practices Score: 95+
- Lighthouse SEO Score: 95+
- First Contentful Paint (FCP): < 1.5s
- Largest Contentful Paint (LCP): < 2.5s
- Time to Interactive (TTI): < 3.5s
- Total Blocking Time (TBT): < 200ms

### Acceptance Criteria
- ✅ All assets minified for production
- ✅ Images optimized and lazy loaded
- ✅ CSS purged of unused Tailwind classes
- ✅ JavaScript code split and conditionally loaded
- ✅ Lighthouse scores meet targets
- ✅ Works with popular WordPress plugins
- ✅ Compatible with all modern browsers
- ✅ Mobile responsive on all devices
- ✅ Passes WAVE accessibility checker
- ✅ No console errors or warnings
- ✅ Fast page load times (<3s on 3G)

#### Cache Busting
- Use `filemtime()` for version numbers
- Automatically updates when files change
- Works better than manual version numbers

#### Critical CSS
- Inline above-the-fold CSS in `<head>`
- Load full CSS asynchronously
- Tools: Critical, Penthouse

#### CDN Integration
- Use WordPress CDN plugins (WP Rocket, etc.)
- Or manually configure CDN URLs
- Test all assets load from CDN

### Dependencies
- All previous phases (1-6)

### References
- [WordPress Performance Best Practices](https://developer.wordpress.org/advanced-administration/performance/optimization/)
- [Lighthouse Documentation](https://developers.google.com/web/tools/lighthouse)
- [WCAG Guidelines](https://www.w3.org/WAI/WCAG21/quickref/)
- `.claude/wordpress-migration-decisions.md` - Performance strategy

### Estimated Effort
2-3 hours

---

**Branch**: `feature/traditional-wordpress-theme`
**Migration**: React SPA → Traditional WordPress Theme
